c-api/compat-5.3.c

```diff
@@ -401,16 +392,22 @@ COMPAT53_API int luaL_fileresult (lua_State *L, int stat, const char *fname) {
   else {
     lua_pushnil(L);
 #if defined(__GLIBC__) || defined(_POSIX_VERSION) || defined(__APPLE__) || (defined(__GNUC__) && (__GNUC__ < 6) && defined (__MINGW32__))
-    // use strerror_r here, because it's available on these specific platforms
+    /* use strerror_r here, because it's available on these specific platforms */
     char buf[512];
-    strerror_r(en, buf, 1024);
+    strerror_r(en, buf, 512);
```

Comment:
strerror_r may not modify the buffer on the XSI-compliant libstdc implementation, so buf needs to be properly initialized.